### PR TITLE
[MLIR][TORCH] Add E2E support for [`aten.mul.Scalar`|`aten.addmm`]

### PR DESCRIPTION
**Pull request by Shukla-Gaurav**
This commit adds lowering of `aten.mul.Scalar` and also adds
decomposition of `aten.addmm` to `aten.mul.Scalar`, `aten.add.Tensor`
and `aten.mm` ops.

Signed-Off-by: Gaurav Shukla <gaurav@nod-labs.com>

### CODE REVIEW

**Review by pashu123 — APPROVED** [review]
LGTM!

**Review by ramiro050 — CHANGES_REQUESTED** [review]
Good job! A few comment below

File: e2e_testing/torchscript/basic.py [hunk 1]
```diff
@@ -105,6 +105,30 @@ def matmul(self, lhs, rhs):
...
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1], torch.float32, True),
+        ([-1, -1], torch.float32, True),
+        ([-1, -1], torch.float32, True),
+    ])
+    def forward(self, M, mat1, mat2):
+        return torch.addmm(M, mat1, mat2, beta=0.0, alpha=1.0)
```
[the remark this answers is not in the document]
Comment (Shukla-Gaurav) [reply e2e_testing/torchscript/basic.py:123]: Thanks, that's a good point. I will update this.

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp [hunk 2]
```diff
@@ -289,6 +289,47 @@ class DecomposeAtenExpandOp : public OpRewritePattern<AtenExpandOp> {
...
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenAddmmOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value input = op.self();
+    Value mat1 = op.mat1();
+    Value mat2 = op.mat2();
+
+    // The operands `self`, `mat1`, `mat2` to aten.addmm must be of rank 2.
+    if (getTensorRank(input) != 2 || getTensorRank(mat1) != 2 ||
```
[the remark this answers is not in the document]
Comment (Shukla-Gaurav) [reply lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:305]: I see. I missed this, will update accordingly.

**Review by cathyzhyi — APPROVED** [review]

**Comment by Shukla-Gaurav** [comment]
Thanks @ramiro050 for spotting broadcastable case. It's fixed now. Can you please review again?

**Comment by ramiro050** [comment]
Awesome! LGTM!

**Pull request merged**